tensorflow/core/kernels/control_flow_ops.cc

```diff
@@ -52,15 +52,9 @@ void SwitchNOp::Compute(OpKernelContext* context) {
   context->set_output(output_index, context->input(0));
 }
 
-REGISTER_KERNEL_BUILDER(
-    Name("Switch").Device(DEVICE_DEFAULT).HostMemory("pred"), SwitchOp);
 REGISTER_KERNEL_BUILDER(
     Name("Switch").Device(DEVICE_TPU_SYSTEM).HostMemory("pred"), SwitchOp);
 
-REGISTER_KERNEL_BUILDER(
-    Name("_SwitchN").Device(DEVICE_DEFAULT).HostMemory("output_index"),
-    SwitchNOp);
-
```

Comment:
Why are we removing these? They should already cover your use case, right? We don't need to modify the GPU registrations.

Reply:
@penpornk This will duplicate with `int32` special kernel and cause a failure : `Multiple Default OpKernel registrations match NodeDef '{{node while/Switch}}': 'op: "Switch" device_type: "DEFAULT" constraint { name: "T" allowed_values { list { type: DT_INT32 } } } host_memory_arg: "data" host_memory_arg: "pred" host_memory_arg: "output_false" host_memory_arg: "output_true"' and 'op: "Switch" device_type: "DEFAULT" host_memory_arg: "pred"'` in `//tensorflow/python/ops/parallel_for:xla_control_flow_ops_test`. I think special registers are more friendly to perfermance.

Reply:
I mean, why do we need to specialize them for int32 since these existing registrations already cover them? We can leave the special int32 registrations for GPU as-is for now (although they could be removed too).

Reply:
But these host arguments restrictions will lost for `PluggableDevice`, and this may have negative impacts on its perfermance.

Reply:
Here we want to make default device to have same logic as GPU device, saying INT32 also as special kernel. If we want to remove GPU’s INT32 special kernel in the future, then we remove them together. Is it ok for you?

Reply:
@penpornk PR has been changed as your request. Since the default case already cover all other cases, any `DEVICE_DEFAULT` register for `Switch` is not needed.